### PR TITLE
feat: smarter market funds

### PR DESCRIPTION
**Pull request by yaohcn**
addBalance = needBalance - availableBalance

### CODE REVIEW

File: chain/deals/client.go [hunk 1]
```diff
@@ -181,7 +182,7 @@ func (c *Client) Start(ctx context.Context, p ClientDealProposal) (cid.Cid, erro
 		smsg, err := c.mpool.MpoolPushMessage(ctx, &types.Message{
 			To:       actors.StorageMarketAddress,
 			From:     p.Client,
-			Value:    types.BigMul(p.PricePerEpoch, types.NewInt(p.Duration)),
+			Value:    types.BigSub(types.BigMul(p.PricePerEpoch, types.NewInt(p.Duration)), clientMarketBalance.Available),
```
Comment (Kubuxu) [inline chain/deals/client.go]: This can go negative, which will error out.

Comment (yaohcn) [reply chain/deals/client.go]: if clientMarketBalance.Available.LessThan(types.BigMul(p.PricePerEpoch, types.NewInt(p.Duration)))

Comment (Kubuxu) [reply chain/deals/client.go]: Aaah, right.

**Comment by magik6k** [comment]
We now have a simple storage market fund manager which handles this

**Pull request closed without merging**